docs/application/dotnet/guides/machine-learning/pipeline.md

```diff
@@ -0,0 +1,181 @@
+
+The main features of Pipeline include:
+
+- [Pipeline construction](#construct)
+
+  You can compose the data stream pipeline through Machine Learning Inference with various elements of [GStreamer](https://gstreamer.freedesktop.org) and [NNStreamer](https://github.com/nnstreamer/nnstreamer).
+
+- [Data flow control](#dataflow)
+
+  With the various nodes and functionalities in Pipeline, you can easily handle and control the data flow.
+
+- [Pushing the input data](#inputdata)
+
+  You can push the input data into the streamed pipeline.
+
+- [Subscribing the output data](#outputdata)
+
+  Pipeline supports multiple output nodes. You can subscribe the output data from the constructed pipeline.
+
+## Prerequisites
+
+To enable your application to use the Machine Learning Inference API functionality:
+
+1. To use the methods and properties of the `Tizen.MachineLearning.Inference.Pipeline` class or its related classes such as `Tizen.MachineLearning.Inference.TensorsData` and `Tizen.MachineLearning.Inference.TensorsInfo`, include the `Tizen.MachineLearning.Inference` namespace in your application.
+
+    ```C#
+    using Tizen.MachineLearning.Inference;
+    ```
+
+2. If the model file you want to use is located in the **media storage** or the **external storage**, the application has to request permission by adding the following privileges to the `tizen-manifest.xml` file.
+
+    ```xml
+    <privileges>
+      <!--To access media storage-->
+      <privilege>http://tizen.org/privilege/mediastorage</privilege>
+
+      <!--To access, read, and write to the external storage-->
+      <privilege>http://tizen.org/privilege/externalstorage</privilege>
+    </privileges>
+    ```
+
+<a name="construct"></a>
+## Pipeline Construction
+
+To construct a pipeline, you first write a pipeline description with [GStreamer](https://gstreamer.freedesktop.org) and [NNStreamer](https://github.com/nnstreamer/nnstreamer) elements.
+Refer to GStreamer manual or NNStreamer documentation for examples and the grammar.
+
+1. Since the model file is located in the resource directory of your own application, you need to get its absolute path.
+
+    ```C#
+    string ResourcePath = Tizen.Applications.Application.Current.DirectoryInfo.Resource;
+    string model_path = ResourcePath + "models/mobilenet_v1_1.0_224_quant.tflite";
+    ```
+
+2. You can construct a pipeline with the description string, including neural network framework and model file.
+
+    ```C#
+    /* Create Pipeline instance with the pipeline description. */
+    string description = "appsrc name=srcx ! " +
+                         "other/tensor,dimension=(string)3:224:224:1,type=(string)uint8,framerate=(fraction)0/1 ! " +
+                         "tensor_filter framework=tensorflow-lite model=" + model_path + " ! " +
+                         "tensor_sink name=sinkx";
+    Pipeline pipe = new Pipeline(description);
+    ```
+
+<a name="dataflow"></a>
+## Data Flow Control
```

Comment:
Data flow control seems a little complicated.
It would be good for beginners to understand to place this at the bottom.

Reply:
Thanks, I updated PR with your opinion.